life_server/Functions/MySQL/fn_bool.sqf

```diff
@@ -1,26 +1,20 @@
 /*
-    File: fn_bool.sqf
-    Author: TAW_Tonic
-
+    File:	fn_bool.sqf
+    Date:   2017-03-22 17:46:27
+    Author: Patrick "Lucian" Schmidt
+    
     Description:
-    Handles bool conversion for MySQL since MySQL doesn't support 'true' or 'false'
-    instead MySQL uses Tinyint for BOOLEAN (0 = false, 1 = true)
+    Converts Bool to Number and Number to bool
 */
-private ["_bool","_mode"];
-_bool = [_this,0,0,[false,0]] call BIS_fnc_param;
-_mode = [_this,1,0,[0]] call BIS_fnc_param;
 
-switch (_mode) do {
-    case 0: {
-        if (_bool isEqualType 0) exitWith {0};
-        if (_bool) then {1} else {0};
-    };
+params[
+    ["_in", "", [0,true]]
+];
+
+if (_in isEqualType true) exitWith {
+    [0,1] select _in;
+};
 
-    case 1: {
-        if (!(_bool isEqualType 0)) exitWith {false};
-        switch (_bool) do {
-            case 0: {false};
-            case 1: {true};
-        };
-    };
+if (_in isEqualType 0) exitWith {
+    [false,true] select _in;
```

Comment:
You're saving ~3μs with the new bool conversion. If it that amount of time mattered to you, then the clarity (and ever so slight improvement) of just using else should as well. It's more for style.

The mres functions were written years ago now, and things such as param(s)/apply/alt syntax to select etc, didn't exist (or were very new). The functions are horrible as they are now, so it'd be worrying if you couldn't make something better.

Regardless, I have purposefully left them, because I plan to remove them, they are not needed (except bool for reduction) - so it's really quite irrelevant at the moment.

If you wish to create a PR/commit, then don't complain about reviews/comments, they are not intended to be nasty or critical (you're free to argue the point if you feel inclined there is good reason). If you wish to make the changes, then I'd appreciate that.

You would also benefit from using arr select bool in your mresArray (with isEqualTo), could probably shave a lot of time off.

Thanks 👍